web/src/lib/req/index.ts

```diff
@@ -0,0 +1,74 @@
+import type { HttpMethod } from '@sveltejs/kit';
+import { Err, Ok, type Option, type Result } from 'rust-optionals';
+
+export type Resp<T extends Record<string, unknown>> = {
+	headers: Headers;
+	body: T;
+};
+
+export const req = async <T extends Record<string, unknown>>(
+	method: HttpMethod,
+	uri: string | URL,
+	body: Option<BodyInit>,
+	headers: Record<string, unknown> = {},
+	fetchOptions: RequestInit = {},
+): Promise<Result<Resp<T>, Error>> => {
+	try {
+		const resp = await fetch(uri, {
+			method,
+			body: body.isNone() ? undefined : JSON.stringify(body.unwrap()),
+			headers: {
+				'Content-Type': 'application/json',
```

Comment:
What happens if you `GET` something with a `Content-Type: application/json` request header and it doesn't return json? just curious

Reply:
Nothing really. If the client is implemented to expect json, and they start parsing the response body as json, they'll get an error that the contents couldn't be parsed as json. They get surprised, checks the raw body, and gets mad at the API for being :hankey: :)

Reply:
The `Content-Type` header is really just the client's way to say, "hey pal, I want a json response". The server is free to ignore that and send back whatever.

The server can do something hella fancy if they respect this header and returns content in several forms per the request :) Imagine some person asking for, ugh, xml, and the server just happily formats the response and sends it back.

There are also some servers that are designed to both be a regular website and an API server, and the client can ask for either on the same endpoint just by passing `application/json` or `text/html`.